mobile/src/screens/ChatScreen.tsx

```diff
@@ -49,122 +58,294 @@ export const ChatScreen: React.FC = () => {
-        <TouchableOpacity onPress={sendMessage}>
-          <Text>send</Text>
+    <SafeAreaView style={styles.contentContainer}>
+      <View style={styles.headerContainer}>
+        <TouchableOpacity onPress={() => setShowMoreHelp(true)}>
+          <Text style={TextStyles.heading2}>{i18nCtx.t('needMoreHelp')}</Text>
         </TouchableOpacity>
       </View>
-      <View style={[styles.container, { display: 'none' }]}>
-        <View style={styles.logoContainer}>
-          <Image
-            source={require('../../assets/images/logo-white.png')}
-            style={styles.logo}
-          />
-        </View>
 
-        <Text style={TextStyles.heading1}>{i18nCtx.t('liveChat')}</Text>
-        <Text style={TextStyles.heading2}>{i18nCtx.t('untilThen')}</Text>
+      <KeyboardAvoidingView
+        style={styles.chatContainer}
+        behavior={Platform.OS == 'ios' ? 'padding' : 'height'}
+      >
+        <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
+          <OfflineIndicator style={{height: '80%'}}>
+            <>
+              {roomId ? (
+                <ScrollView
+                  showsVerticalScrollIndicator={false}
+                  ref={messagesViewRef}
+                  onContentSizeChange={() =>
+                    (
+                      messagesViewRef.current as unknown as ScrollView
+                    )?.scrollToEnd({
+                      animated: true,
+                    })
+                  }
+                >
+                  {messages.map(({ id, text, from }, index) => {
+                    if (auth.user)
+                      return (
+                        <View key={id}>
+                          {from !== auth.user.name &&
+                          messages[index - 1].from !== from ? (
+                              <Text style={styles.senderNameText}>{from}</Text>
+                            ) : null}
+                          <View
+                            style={[
+                              styles.mainChatBubble,
+                              from === auth.user.name
+                                ? styles.rightChatBubble
+                                : styles.leftChatBubble,
+                            ]}
+                          >
+                            <Text
+                              style={[
+                                TextStyles.caption2,
+                                {
+                                  color:
+                                    from === auth.user.name
+                                      ? Colors.white
+                                      : Colors.shadowColor,
+                                },
+                              ]}
+                            >
+                              {text}
+                            </Text>
+                          </View>
+                        </View>
+                      );
+                  })}
```

Comment:
Can you clean this up into a `ChatBubble` component please? It's a little hard to follow atm